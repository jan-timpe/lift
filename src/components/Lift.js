import React, { Component } from 'react'
import '../style/Lift.css'
import LiftResultList from './LiftResultList'
import LiftResultForm from './LiftResultForm'
import { connect } from 'react-redux'
import { addLiftResult } from '../actions'


const mapStateToProps = (state, ownProps) => ({
  results: state.liftResults.filter((result) => (result.liftId == ownProps.lift.id))
})

const mapDispatchToProps = (dispatch) => ({
  addLiftResult: args => dispatch(addLiftResult(args))
})


class Lift extends Component {

  constructor(props) {
    super(props)

    this.state = {
      weight: '',
      expanded: false,
    }
  }

  getMaxWeight() {
    var max = null
    let results = this.props.results;
    for(var i = 0; i < results.length; i++) {
      if(results[i].weight > max) {
        max = results[i].weight
      }
    }
    return max
  }

  toggleExpanded() {
    this.setState({ expanded: !this.state.expanded })
  }

  render() {

    let expandedCell = null;
    if(this.state.expanded) {
      expandedCell = (
        <div className="expanded">
          <LiftResultForm addLiftResult={this.props.addLiftResult} liftId={this.props.lift.id} />
          <LiftResultList results={this.props.results} />
        </div>
      )
    }

    return (
      <div className="Lift">
        <div className="clickable" onClick={ () => {
          this.toggleExpanded()
        }}>
          <div className="name">
            { this.props.lift.name }
          </div>
          <div className="max-weight">
            { this.props.results.length > 0 ? this.getMaxWeight() : null }
          </div>
        </div>

          { expandedCell }

      </div>
    );
  }


}


export default connect(mapStateToProps, mapDispatchToProps)(Lift)