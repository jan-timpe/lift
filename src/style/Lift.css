.lift {
  width: 100%;
  background: #fff;
  color: #000;
  margin: 0px 4px;
  border-bottom: 4px solid #ddd;
}

.lift .name, .lift .max-weight {
  padding: 12px;
  font-size: 18pt;
  display: inline-block;
}

.lift:hover {
  cursor: pointer;
}

.max-weight {
  background: #66638B;
  float: right;
  text-align: right;
  color: #fff;
  font-family: 'Menlo', monospace;
}

.expanded {
  background: #ddd;
}

.expanded .result {
  padding: 8px;
  font-size: 12pt;
  border-bottom: 2px solid #aaa;
}

.expanded label {
  color: #000;
}