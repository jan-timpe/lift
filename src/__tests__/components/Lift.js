import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'
import Lift from '../../components/Lift'

const mockStore = configureStore()

const initialState = {
  lifts: [{
    id: 0,
    name: 'Squat'
  }, {
    id: 1,
    name: 'Bench'
  }],
  liftResults: [{
    id: 0,
    liftId: 0,
    weight: 205,
    date: '7/1/2018'
  }]
}

const store = mockStore(initialState)

describe(`Lift`, () => {

  // snapshot
  it(`should render`, () => {
    const wrapper = shallow(
      <Lift store={store} lift={store.getState().lifts[0]} />)
    const component = wrapper.dive()
    expect(toJson(component)).toMatchSnapshot()
  })


  // expands when clicked

  it(`should hide expanded when clicked`, () => {
    const wrapper = shallow(
      <Lift store={store} lift={store.getState().lifts[0]} />)
    const component = wrapper.dive()

    expect(component.find('.expanded').exists()).toEqual(false)
    
  })

  it(`should show expanded when clicked`, () => {
    const wrapper = shallow(
      <Lift store={store} lift={store.getState().lifts[0]} />)
    const component = wrapper.dive()
    
    component.find('.clickable').simulate('click')
    expect(component.find('.expanded').exists()).toEqual(true)
  })

})