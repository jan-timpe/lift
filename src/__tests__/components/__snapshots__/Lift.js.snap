// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lift should render 1`] = `
<div
  className="Lift"
>
  <div
    className="clickable"
    onClick={[Function]}
  >
    <div
      className="name"
    >
      Squat
    </div>
    <div
      className="max-weight"
    >
      205
    </div>
  </div>
</div>
`;
